### PR TITLE
Allow lilac to be installed as a system package

### LINKED ISSUE #35: write a setup.py

### PR DESCRIPTION
**Pull request by yan12125**
Based on #78. Fixes #35. For archlinuxcn/repo#1651.

Some implementation notes

* I use setuptools-scm to generate the Python package version
* The Python package name is archlinuxcn-lilac (https://github.com/archlinuxcn/lilac/issues/35#issuecomment-662251124)
* `~/.lilac/config.ini` is another path for the configuration file
* A proof-of-concept PKGBUILD can be found at https://github.com/yan12125/repo/tree/3d74b0b64d5c9aaa20f89aa1bd38e86969a59fae/archlinuxcn/lilac-git
* `config.ini.sample` is not handled by `setup.py`. I believe it should be installed in /usr/share/doc. However, someone may want to install this package with `python setup.py install --user`, so I leave it to downstream packagers. See PKGBUILD above as an example.

Tested scenarios:
* `./lilac lilac-git` from a git clone
* `lilac lilac-git` with the package created by PKGBUILD above

/cc @Xuanwo @oldherl

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,22 @@ __pycache__/
...
+
+# Distribution / packaging
+.Python
+build/
+develop-eggs/
+dist/
+downloads/
+eggs/
+.eggs/
+lib/
```
Comment (lilydjwg) [inline .gitignore:15]: Some of these are pointless. Please don't add entries defensively. Changes are expected to be reviewed before committing so any unintential additions should be noticed.

Comment (yan12125) [reply .gitignore:15]: Maybe just remove those entries? Some of those entries are for generated file from `python setup.py install`, `python setup.py sdist` or `python setup.py build`. But I guess those setup.py commands will not appear during development of lilac?

Comment (lilydjwg) [reply .gitignore:15]: `setup.py` generates only `dist`, `sdist` and `build`. I don't know where others come from.

Comment (Xuanwo) [reply .gitignore:15]: They are generated by IDEA and cloud be deleted if not used.

Comment (yan12125) [reply .gitignore]: `*.egg-info` is generated from `python setup.py egg_info` (a command used by `python setup.py build` or `python setup.py install`).
`.eggs` is used to store packages specified in `setup_requires` if they are not installed beforehand.

In the updated commit (6f2479f8d9ac2e088bda62112a783bf63b08ef1b), I keep entries that might be generated by setuptools and remove others.

**Pull request merged**